File: app/src/main/java/com/iceteaviet/fastfoodfinder/data/prefs/PreferencesHelper.java
package com.iceteaviet.fastfoodfinder.data.prefs;

/**
 * Created by tom on 7/24/18.
 */
public interface PreferencesHelper {
    void putString(String key, String value);

    String getString(String key, String defaultValue);

    void putBoolean(String key, Boolean value);

    Boolean getBoolean(String key, Boolean defaultValue);

    Boolean getAppLaunchFirstTime();

    void setAppLaunchFirstTime(Boolean isFirstTime);

    int getNumberOfStores();

    void setNumberOfStores(int numberOfStores);

    String getCurrentUserUid();

    void setCurrentUserUid(String uid);
}

File: app/src/main/kotlin/com/iceteaviet/fastfoodfinder/data/prefs/PreferencesHelper.kt
package com.iceteaviet.fastfoodfinder.data.prefs

/**
 * Created by tom on 7/24/18.
 */
interface PreferencesHelper {

    fun putString(key: String, value: String)

    fun getString(key: String, defaultValue: String): String

    fun putBoolean(key: String, value: Boolean)

    fun getBoolean(key: String, defaultValue: Boolean): Boolean

    fun setStringSet(key: String, set: MutableSet<String>)

    fun getStringSet(key: String, defaultValue: MutableSet<String>): MutableSet<String>

    fun getAppLaunchFirstTime(): Boolean

    fun setAppLaunchFirstTime(isFirstTime: Boolean)

    fun getNumberOfStores(): Int

    fun setNumberOfStores(numberOfStores: Int)

    fun getSearchHistories(): MutableSet<String>

    fun setSearchHistories(set: MutableSet<String>)
}
